src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,113 @@
 package core.basesyntax;
 
-public class MyHashMap<K, V> implements MyMap<K, V> {
+import java.util.Objects;
+
+public class MyHashMap<K,V> implements MyMap<K,V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTORY = 0.75f;
+
+    private Node<K,V>[] mapTable;
+    private int size;
+    private final float loadFactor;
+    private int threshold;
+
+    @SuppressWarnings("unchecked")
+    public MyHashMap() {
+        this.loadFactor = DEFAULT_LOAD_FACTORY;
+        this.threshold = (int) (DEFAULT_LOAD_FACTORY * DEFAULT_CAPACITY);
+        mapTable = (Node<K, V>[]) new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int hash = hashCode(key);
+        int index = (mapTable.length - 1) & hash;
+        Node<K,V> specificNode = mapTable[index];
+        if (specificNode == null) {
+            mapTable[index] = new Node<>(hash, key, value, null);
+        } else {
+            Node<K,V> tempNode;
+            if (specificNode.hash == hash && (Objects.equals(key, specificNode.key))) {
+                tempNode = specificNode;
+            } else {
+                for ( ; ; ) {
+                    if ((tempNode = specificNode.next) == null) {
+                        specificNode.next = new Node<>(hash, key, value, null);
+                        break;
+                    }
+                    if (tempNode.hash == hash && (tempNode.key == key
+                            || (key != null && key.equals(specificNode.key)))) {
+                        break;
+                    }
+                    specificNode = tempNode;
+                }
+            }
+            if (tempNode != null) {
+                tempNode.value = value;
+                return;
+            }
+        }
+        if (++size > threshold) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
+        int hash = hashCode(key);
+        Node<K,V> node = mapTable[(mapTable.length - 1) & hash];
+        Node<K,V> nodeNext;
+        if (node != null) {
+            if (Objects.equals(key, node.key)) {
```

Comment:
Make it simpler
![image](https://user-images.githubusercontent.com/20326858/226898539-999e5495-fd16-42e8-8e1d-f9c8c86498e4.png)